<launch>
	<rosparam param="use_sim_time">true</rosparam>
	<arg name="rviz" default="true"/>
	<arg name="rate" default="1"/>
	<arg name="apriltag" default="false"/>
	<arg name="bag_dir" value="$(find irols)/bags"/>
	<arg name="rosbag_file" value="$(arg bag_dir)/estimate_data_vision_only.bag" unless="$(arg apriltag)"/>
	<arg name="rosbag_file" value="$(arg bag_dir)/estimate_data_camfix.bag" if="$(arg apriltag)"/>

	<include file="$(find irols)/launch/ekf.launch"/>

	<node name="vision_estimator" pkg="irols" type="vision_estimator.py"/>
	<node name="estimator_transform" pkg="irols" type="estimator_transforms.py"/>

	<node name="rosbag_replay" pkg="rosbag" type="play" args="--clock $(arg rosbag_file) --rate $(arg rate)"/>
	<node pkg="rviz"
		  type="rviz"
		  name="rviz"
		  args="-d $(find irols)/configs/estimator_validation.rviz"
		  if="$(arg rviz)"
		  />
	<group if="$(arg apriltag)">
		<include file="$(find irols)/launch/image_pipeline.launch"/>
		<include file="$(find irols)/launch/apriltags_ros.launch"/>
		<node name="apriltag_estimator" pkg="irols" type="apriltag_estimator.py"/>
	</group>
</launch>
